internal/controller/api/v1alpha1/cluster_types.go

```diff
@@ -31,18 +31,18 @@ type ClusterSpec struct {
 	// Installation will start once cluster is in known state and approved
 	Approved                 bool   `json:"approved,omitempty"`
 	OpenshiftVersion         string `json:"openshiftVersion"`
-	BaseDnsDomain            string `json:"baseDnsDomain,omitempty"`
-	ClusterNetworkCidr       string `json:"clusterNetworkCidr,omitempty"`
+	BaseDNSDomain            string `json:"baseDNSDomain,omitempty"`
+	ClusterNetworkCIDR       string `json:"clusterNetworkCIDR,omitempty"`
 	ClusterNetworkHostPrefix int64  `json:"clusterNetworkHostPrefix,omitempty"`
-	ServiceNetworkCidr       string `json:"serviceNetworkCidr,omitempty"`
-	ApiVip                   string `json:"apiVip,omitempty"`
-	ApiVipDnsName            string `json:"apiVipDnsName,omitempty"`
-	IngressVip               string `json:"ingresVip,omitempty"`
-	MachineNetworkCidr       string `json:"machineNetworkCidr,omitempty"`
-	SshPublicKey             string `json:"sshPublicKey,omitempty"`
-	VipDhcpAllocation        bool   `json:"vipDhcpAllocation,omitempty"`
-	HttpProxy                string `json:"httpProxy,omitempty"`
-	HttpsProxy               string `json:"httpsProxy,omitempty"`
+	ServiceNetworkCIDR       string `json:"serviceNetworkCIDR,omitempty"`
+	APIVIP                   string `json:"APIVIP,omitempty"`
```

Comment:
Now we're getting picky... :)

I think the more important aspect of the convention is that the json field name start lower-case. That applies even to acronyms, and even to acronyms followed by other stuff. For example, the Volume resource has a field `awsElasticBlockStore`. Having two acronyms back-to-back is more unusual, but I think the right thing to do would be `apiVIP`.

Reply:
Thanks, Michael. Updated `JSON` names accordingly.